### ISSUE TITLE
Allow fast_track to read mappings

### ISSUE THREAD

**Issue by engn33r** [rainshowerLabs/sothis#18]
**Is your feature request related to a problem? Please describe.**

fast_track reads state variables. Some state variables are not as easy to read as uint256 variables. For example, consider price data provided by Chainlink. It can be requested with `latestAnswer()`, but the source data is stored in a uint32 -> Transmission mapping. If sothis can read from mappings, it will be easy to pull historic token price data directly from the blockchain without any data brokers.

![img](https://github.com/rainshowerLabs/sothis/assets/6261182/3f9b74ad-da80-4277-9453-1b04e454f337)

**Describe the solution you'd like**

Read data from mappings. A new flag will be needed to identify which item in the mapping to read from. Handling of custom structs will be needed, likely with another flag which passes the struct layout/ABI.

**Describe alternatives you've considered**

None

**Additional context**

None

**Comment by makemake-kbo** [comment]
this would be very interesting but also probably very tedious

**Comment by makemake-kbo** [comment]
this isnt that tedious but is not going to be appearing in sothis anytime soon.

For now, im working on a new tracking mode to track eth_call/staticcall outputs

**Comment by engn33r** [comment]
Agreed, not a priority. For reference, `cast index` enables computing of a mapping value's storage slot (in solidity, not [in vyper](https://github.com/a16z/halmos/issues/154#issuecomment-1673998226)). And I was able to manually determine the storage slot for the chainlink mainnet ETH/USD price, it can be queried with `cast --to-dec (cast storage 0x50ce56a3239671ab62f185704caedf626352741e 0x6200d9c0606964564958e1edf0b5d67e485691a223ce33e3267306cb461a390e)`
